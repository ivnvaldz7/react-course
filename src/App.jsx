import PropTypes from "prop-types";

export const App = ({ title, subtitle, name }) => {
  return (
    <>
      <h1>{title}</h1>
      <h2>{subtitle}</h2>
      <h4>{`Con ${name}`}</h4>
    </>
  );
};
App.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  name: PropTypes.string.isRequired,
};
App.defaultProps = {
  subtitle: "No existe texto",
  name: "No existe texto",
};
