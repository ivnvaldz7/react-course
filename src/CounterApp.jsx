import PropTypes from "prop-types";
import { useState } from "react";
export const CounterApp = ({ value, title }) => {
  const [counter, setCounter] = useState(value);
  return (
    <>
      <h1>{title}</h1>
      <h2>{counter}</h2>
      <button onClick={() => setCounter(counter + 1)}> +1</button>
      <button onClick={() => setCounter(counter - 1)}> -1</button>
      <button onClick={() => setCounter(value)}> reset</button>
    </>
  );
};
CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
  title: PropTypes.string
};
CounterApp.defaultProps = {
  title: "Title of Counter"
}
