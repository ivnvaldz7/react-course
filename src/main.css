body,html{
    background-color: #21232A;
    color: aliceblue;
    font-family: helvetica, arial;
    font-size: 1rem;
    padding: 70px;
}
h4{
    font-family: 'Italic';
}
h1{
    font-weight: Bold;
}
p{
    font-size: 2rem;
    font-weight: Bold;
}
button{
    padding: 1rem;
    outline: none;
    border-radius: 15px;
    margin: 1rem;
    font-size: 1rem;
}