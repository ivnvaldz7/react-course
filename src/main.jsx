import React from "react";
import ReactDOM from "react-dom/client";
import { CounterApp } from "./CounterApp";
import {App} from './App';
import "./main.css";
const root = ReactDOM.createRoot(document.getElementById("root"));

root.render(
  <React.StrictMode>
    <CounterApp value={10} />
    {/* <App title="Curso de React" subtitle='De 0 a experto' name='Fernando Herrera'/> */}
  </React.StrictMode>
);
