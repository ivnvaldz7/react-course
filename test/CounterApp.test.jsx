import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe("Pruebas en CounterApp", () => {
    const value = 10;
  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<CounterApp value={value} />);
    expect(container).toMatchSnapshot();
  });
  test("Debe verificar que el counter inicial sea 100", () => {
    render(<CounterApp value={value} />);
    //Tarea: validar que el value sea el que se necesita
    expect(screen.getByText(value)).toBeTruthy();
    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain('10');
  });
  test('Simular clicks en el boton +1', () => { 
    render(<CounterApp value={value} />);
    fireEvent.click(screen.getByText('+1'))
    expect(screen.getByText('11')).toBeTruthy()
   })
   test('Simular clicks en el boton +1', () => { 
    render(<CounterApp value={value} />);
    fireEvent.click(screen.getByText('-1'))
    expect(screen.getByText('9')).toBeTruthy()
   })
});
