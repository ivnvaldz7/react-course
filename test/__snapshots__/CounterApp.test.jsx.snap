// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Title of Counter
  </h1>
  <h2>
    10
  </h2>
  <button>
     +1
  </button>
  <button>
     -1
  </button>
  <button>
     reset
  </button>
</div>
`;
